packages/http/src/experimental/merge-patch/helpers.ts

```diff
@@ -0,0 +1,76 @@
+import { isArrayModelType, Model, ModelProperty, Program, Type, Value } from "@typespec/compiler";
+import { HttpStateKeys } from "../../lib.js";
+
+/**
+ * Determines if the given model is part of a mergePatch transform
+ * @param program The compiled TypeSpec program
+ * @param model The model to check
+ * @returns true if the model was generated using a mergePatch template, otherwise false
+ */
+export function isMergePatch(program: Program, model: Model): boolean {
+  return program.stateMap(HttpStateKeys.mergePatchModel).has(model);
```

Comment:
for new decorators you can use `useStateSet` or `useStateMap` to consturct typed state accessors